pkgs/misc/emulators/caprice32/default.nix

```diff
@@ -16,6 +19,8 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ desktop-file-utils pkg-config ];
   buildInputs = [ libpng SDL freetype zlib ];
 
+  patches = [ ./string.patch ];
```

Comment:
Is possible to use fetchpatch to fetch the upstream commit fixing this? nixpkgs prefers fetching patches to adding them inline.

Reply:
Thanks for the hint. I tried replacing the patches line (also adding `fetchpath` to the top parameter line) with:

```
  patches = [
    (fetchpatch {
      url = "https://github.com/ColinPitrat/caprice32/commit/6b250ead2a8b66ed2b676124368191e7adc11768.patch";
      sha256 = "";
    })
  ];
```

However, trying to build this yields me:

```
> nix build /home/jan/Projects/nix/nixpkgs#caprice32
warning: Git tree '/home/jan/Projects/nix/nixpkgs' is dirty
warning: found empty hash, assuming 'sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA='
error: builder for '/nix/store/4hbq2mz4mczkj9572xfq6g8hpy9wg9b0-6b250ead2a8b66ed2b676124368191e7adc11768.patch.drv' failed with exit code 1;
       last 7 log lines:
       >
       > trying https://github.com/ColinPitrat/caprice32/commit/6b250ead2a8b66ed2b676124368191e7adc11768.patch
       >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
       >                                  Dload  Upload   Total   Spent    Left  Speed
       > 100 25305  100 25305    0     0    99k      0 --:--:-- --:--:-- --:--:--   99k
       > /nix/store/p799a5y9pkw2rqs1s83f5aah7ds4pkr9-stdenv-linux/setup: line 88: /build/: Is a directory
       > sed: couldn't flush stdout: Broken pipe
       For full logs, run 'nix log /nix/store/4hbq2mz4mczkj9572xfq6g8hpy9wg9b0-6b250ead2a8b66ed2b676124368191e7adc11768.patch.drv'.
error: 1 dependencies of derivation '/nix/store/za4gvf38y4y06wp4fy449141yy4q1vji-caprice32-4.6.0.drv' failed to build
```

I'm quite new with Nix and have no idea what might be wrong at this point or how to debug this. Looking at the whole log also does not give any enlightenment.

Reply:
Figured it out, I need to put `lib.fakeSha256` instead of an empty hash. I will prepare an update.

Reply:
Note: if you use edits, we don't get notifications from github. Please reply instead.